library/Icingadb/Widget/ViewModeSwitcher.php

```diff
@@ -104,15 +123,28 @@ protected function assemble()
         $currentViewMode = $this->getViewMode();
 
         foreach (static::$viewModes as $viewMode => $icon) {
-            $url = $this->url->with($viewModeParam, $viewMode);
-
-            $link = Html::tag('li', new Link(new IcingaIcon($icon), $url));
+            $protectedId = $this->protectId($icon);
```

Comment:
The id is now protected, fine, but there's still no `view-mode-switcher-` prefix! It doesn't matter if protected or not, the icon name isn't unique enough for being used as an id.